pyaerocom/helpers.py

```diff
@@ -1682,3 +1684,73 @@ def get_time_rng_constraint(start, stop):
     t_upper = iris.time.PartialDateTime(year=stop.year, month=stop.month, day=stop.day)
 
     return iris.Constraint(time=lambda cell: t_lower <= cell <= t_upper)
+
+
+def get_max_period_range(periods):
+    start = min([int(per.split("-")[0]) for per in periods])
+    stop = max(int(per.split("-")[1]) if len(per.split("-")) > 1 else int(per) for per in periods)
```

Comment:
you can simplify the generator expression by processing the last element on the split,
if you are concerned about invalid periods you could add an assert

``` python
    assert all(per.count("-") <= 1 for per in periods), f"invalid period in {periods=}"
    stop = max(int(per.split("-")[-1]) for per in periods)
```